packages/cadl-ranch-expect/src/decorators.ts

```diff
@@ -133,8 +133,26 @@ function getRouteSegments(program: Program, target: Operation | Interface | Name
 }
 
 function getOperationRoute(program: Program, target: Operation): string {
+  const template = getHostTemplate(program, target.namespace);
   const segments = getRouteSegments(program, target);
-  return "/" + segments.map((x) => (x.startsWith("/") ? x.substring(1) : x)).join("/");
+  return (template ? template : "/") + segments.map((x) => (x.startsWith("/") ? x.substring(1) : x)).join("/");
+}
+
+function getHostTemplate(program: Program, namespace?: Namespace): string | undefined {
+  if (namespace === undefined) {
+    return undefined;
+  }
+  const server = getServers(program, namespace);
+  if (server && server.length === 1) {
+    if (server[0].url.indexOf("localhost:3000") > -1) {
+      return server[0].url.split("localhost:3000")[1];
+    } else if (server[0].url.indexOf("{endpoint}") > -1) {
+      return server[0].url.split("{endpoint}")[1];
+    } else {
+      return server[0].url;
+    }
+  }
```

Comment:
I am changing it this way.

Reply:
@timotheeguerin one reason I kind of don't like URL is, it automatically encoding something 
<img width="896" alt="image" src="https://github.com/Azure/cadl-ranch/assets/55688292/ec147bb7-3022-43b8-8490-0887800fd4da">

Reply:
The above is the result I get after I change the code like 
```typescript
  if (server && server.length === 1) {
    try {
      return new URL(server[0].url).pathname;
    } catch (e) {
      if (server[0].url.indexOf("{endpoint}") > -1) {
        return server[0].url.split("{endpoint}")[1];
      } else {
        return server[0].url;
      }
    }
  }
```

Reply:
1. It is not actually "host". Seems more like `getRouteSegmentFromServer`?
2. Hope there is more generic solution for `localhost:3000` and `{endpoint}`, but I don't have good solution. Maybe remove URL scheme and start with first `/`, but I guess there will be bug on this as well.

Reply:
yeah I guess that's ok we can revisit if we get into more cases that don't work out

Reply:
changed the name to getRouteSegmentFromServer.